### ISSUE TITLE
Possible memory leak when using ENGINE_by_id() with tpm2tss as argument

### ISSUE THREAD

**Issue by bjaggi-sag** [openssl/openssl#20349]
I have customised the Paho Mqtt C library to accept Engine name as an argument. When i use **tpm2tss** engine as an argument with **ENGINE_by_id()** with **OpenSSL 1.1.1f** we get possisble memory leaks in the Valgrind report. 
Below is the Memory leak summary:

 ```
==2619092== LEAK SUMMARY:
==2619092==    definitely lost: 0 bytes in 0 blocks
==2619092==    indirectly lost: 0 bytes in 0 blocks
==2619092==      possibly lost: 2,992 bytes in 27 blocks
==2619092==    still reachable: 122,589 bytes in 1,236 blocks
==2619092==                       of which reachable via heuristic:
==2619092==                         length64           : 968 bytes in 17 blocks
==2619092==                         newarray           : 1,728 bytes in 28 blocks
==2619092==         suppressed: 0 bytes in 0 blocks
 ```
and the specific call stack that initiates from Engine_by_id is below:

 ```
==2619092== 8 bytes in 1 blocks are still reachable in loss record 8 of 1,078
==2619092==    at 0x483B7F3: malloc (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==2619092==    by 0x4B3231A: CRYPTO_strdup (in /usr/lib/x86_64-linux-gnu/libcrypto.so.1.1)
==2619092==    by 0x4AFFB5B: ??? (in /usr/lib/x86_64-linux-gnu/libcrypto.so.1.1)
==2619092==    by 0x4AFF49A: ENGINE_ctrl_cmd_string (in /usr/lib/x86_64-linux-gnu/libcrypto.so.1.1)
==2619092==    by 0x4B018AE: ENGINE_by_id (in /usr/lib/x86_64-linux-gnu/libcrypto.so.1.1)
==2619092==    by 0x498B415: SSLSocket_createContext (in /usr/local/lib/libpaho-mqtt3as.so.1.3.9)
==2619092==    by 0x498B985: SSLSocket_setSocketForSSL (in /usr/local/lib/libpaho-mqtt3as.so.1.3.9)
==2619092==    by 0x4989741: MQTTAsync_connecting (in /usr/local/lib/libpaho-mqtt3as.so.1.3.9)
==2619092==    by 0x498A012: MQTTAsync_cycle (in /usr/local/lib/libpaho-mqtt3as.so.1.3.9)
==2619092==    by 0x498747A: MQTTAsync_receiveThread (in /usr/local/lib/libpaho-mqtt3as.so.1.3.9)
==2619092==    by 0x4C8B608: start_thread (pthread_create.c:477)
==2619092==    by 0x51DD132: clone (clone.S:95)
 ```

I am doing the Engine_free(engineName) to free up the resources allocated by Engine_by_id() but still its showing possibly lost memory.

**Comment by paulidale** [comment]
I'm only seeing two strdup calls in the tpm2tss engine and both look like they will be cleaned up correctly.

Is there anything on the error stack when this happens?
Could you try a debug build which will provide more information.

**Comment by nhorman** [comment]
marking as inactive, will be closed at the completion of 3.4 dev without further input
